### DISCUSSION TITLE
Parallel static export

### DISCUSSION THREAD

**Discussion by umarmw** [vercel/next.js#14929] [Help]
Hello,

I have a site which has 800 article items from contentful and I'm generating the site statically and deploy on Azure. It takes around 12 mins for the `next build && next export` command to complete. Is there a way to parallelize this?

Thanks

**Comment by umarmw** [comment]
Hello guys,

is there a way to split the `next export` task in smaller batches?

Thanks

**Reply by timneutkens** [reply]
Next.js already utilizes all available cores of the build machine

**Reply by banshee** [reply]
It'd be useful to be able to split it across multiple machines.  Could next export take a list of paths, and then have a way to combine the results (even just with something like rsync)?  Things like https://www.gnu.org/software/parallel/ make it really easy to have build farms when your commands just take lists of files, and things like AWS make those build farms really inexpensive.  (I've got a site I'm working on that potentially has 17k or so individual pages for content snippets, and there's no real reason they can't just be generated in advance and not at request time.  And 17k pages is still well within what most people would call a tiny-to-small site - what about sites with a few hundred k products?  That's still fast and easy to serve as static files, but painful to generate on a single machine.)

**Reply by maksimgorodilov** [reply]
But does it parallelize going over static parameters for SSG? So then we can utilize network channel size completely?
